api/grpcserver.go

```diff
@@ -337,9 +338,29 @@ func (svr *GRPCServer) GetRawBlocks(ctx context.Context, in *iotexapi.GetRawBloc
 
 // GetLogs get logs filtered by contract address and topics
 func (svr *GRPCServer) GetLogs(ctx context.Context, in *iotexapi.GetLogsRequest) (*iotexapi.GetLogsResponse, error) {
-	ret, err := svr.coreService.Logs(in)
+	if in.GetFilter() == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty filter")
+	}
+	var (
+		ret []*iotextypes.Log
+		err error
+	)
+	switch {
+	case in.GetByBlock() != nil:
+		var blkHeight uint64
+		blkHeight, err = svr.coreService.dao.GetBlockHeight(hash.BytesToHash256(in.GetByBlock().BlockHash))
```

Comment:
we'd better not to use private variable of coreService

Reply:
We could add a new func `GetBlockHeight()` in another pr

Reply:
it should be LogsByBlockHash

Reply:
BlockMetaByHash() also shares the dao.GetBlockHeight() func